### PR TITLE
Refact/http conv

### PR DESCRIPTION
**Pull request by nejisama**
Refactory default HTTP1 and HTTP2 convert, remove the proxy convert, use transcoder filter instead.

The streamfilter `mirror` will not allowed protocol convert. 

TODO: remove the stream filter api `SetConvert`

### CODE REVIEW

File: pkg/proxy/streamfilters.go [hunk 1]
```diff
@@ -130,7 +127,6 @@ func (f *streamReceiverFilterHandler) SendHijackReplyWithBody(code int, headers
 
 func (f *streamReceiverFilterHandler) SendDirectResponse(headers types.HeaderMap, buf types.IoBuffer, trailers types.HeaderMap) {
 	atomic.StoreUint32(&f.activeStream.reuseBuffer, 0)
-	f.activeStream.noConvert = true
```
Comment (taoyuanyuan) [inline pkg/proxy/streamfilters.go:133]: transcoder在SendDirectResponse场景下行为是否一致？

Comment (nejisama) [reply pkg/proxy/streamfilters.go:133]: SendDirectResponse和SendHijack 的场景，TranscodingResponse收到应该都不是预期以内的响应。对转换实现新增类型判断，如果类型判断不就就原样返回保持兼容。

**Review by taoyuanyuan — APPROVED** [review]
LGTM

**Review by wangfakang — APPROVED** [review]
LGTM

**Pull request merged**